core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -404,8 +407,14 @@ public void addEnv(String key, String value) {
         env.add(key + "=" + value);
     }
 
+    public void addHostSystemAwareFileSystemBind(String hostPath, String containerPath, BindMode mode) {
```

Comment:
since we have an "if" statement, I think it makes sense to add it to `addFileSystemBind`. It will not break the old code because previously Windows was unsupported (at least officially), but will remove the need to rewrite all test suits replacing "addFileSystemBind" with "addHostSystemAwareFileSystemBind"